### PR TITLE
Port to dune and new pb

### PR DESCRIPTION
**Pull request by rgrinberg**
* Get rid of ocamlbuild in favor of dune
* Use wrapped names everywhere
* Remove topkg cruft

This is the preliminary PR before I merge the 2 projects.

### CODE REVIEW

**Comment by rgrinberg** [comment]
The tests are still failing:

```
File "/Users/rgrinberg/github/yallop/ocaml-pb-plugin/_build/default/test/oUnit-Protobuf wire format tests-Rudis-MacBook-Pro.local#01.log", line 2, characters 1-1:
Error: Protobuf wire format tests:0:test interoperability (small read) (in the log).

Raised by primitive operation at file "stdlib.ml", line 398, characters 28-54
Called from file "src/batFile.ml", line 144, characters 49-82
Called from file "src/batFile.ml", line 148, characters 13-21
Called from file "test/test.ml", line 27, characters 21-52
Called from file "src/lib/ounit2/advanced/oUnitRunner.ml", line 83, characters 13-26

Sys_error("small.python.serialized: No such file or directory")
```

How did they work before? I don't see any code that generated these files for python.

**Comment by yallop** [comment]
`small.python.serialized` is generated by [the `open` call in `test_gen.py`](https://github.com/yallop/ocaml-pb-plugin/blob/53a93bea8ac37d7825f14e5aaa63cc05efb9490d/test/test_gen.py#L75).  The [comments in the `test` target in the `Makefile`](https://github.com/yallop/ocaml-pb-plugin/blob/53a93bea8ac37d7825f14e5aaa63cc05efb9490d/Makefile#L14-L31) outline the steps involved.

**Comment by rgrinberg** [comment]
> small.python.serialized is generated by the open call in test_gen.py. The comments in the test target in the Makefile outline the steps involved.

Thanks for the pointer. That's quite similar to the test setup in ocaml-pb. Merging the repos will make it easier to share these fixtures.

File: src/protobuf_reified.ml [hunk 1]
```diff
@@ -76,11 +76,11 @@ struct
...
-    method_name: string;
-    input_type: string;
-    output_type: string;
-  }
+  (* and method_ = {
+   *   method_name: string;
+   *   input_type: string;
+   *   output_type: string;
+   * } *)
   and basic = {
```
Comment (yallop) [inline src/protobuf_reified.ml]: Why is `method_` commented out?  If it's due to a warning arising from `dune`'s defaults, let's disable the warning locally instead.

File: test/test.ml [hunk 2]
```diff
@@ -115,7 +116,8 @@ let test_interoperability_small_write _ =
 
 
 let test_interoperability_comprehensive_write _ =
-  let u32 = UInt32.of_int and u64 = UInt64.of_int and bytes = Bytes.of_string in
+  let u32 = Unsigned.UInt32.of_int
+  and u64 = Unsigned.UInt64.of_int and bytes = Bytes.of_string in
```
Comment (yallop) [inline test/test.ml]: Please move `and bytes = Bytes.of_string in` to the next line.

**Comment by yallop** [comment]
I think it'd be useful for discoverability to leave a tiny `Makefile` that just invokes `dune` in place.

**Comment by rgrinberg** [comment]
Do you mind if I address your comments in the PR that merges this repo? It will save me some intermediate work this way.

**Comment by rgrinberg** [comment]
I've addressed your comments in the PR that imports this repository.

**Pull request merged**